lua updater: support in-repo rockspecs

teto opened the pull request:
###### Motivation for this change
Currently the lua updater only considers packages registered on luarocks.org. If we want to support lua-based vim plugins that dont release on luarocks, this is needed.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

Commit:
update-luarocks-packages: support in-repo rockspecs

to handle derivations that are not registered or updated on
luarocks.org, for instance neovim plugins.

Commit:
update-luarocks-packages: use the current nixpkgs

restore the behavior of the previous updater that was using the current revision of nixpkgs instead of an outer one

Commit:
luaPackages: update

lblasc commented on maintainers/scripts/luarocks-packages.csv:
```diff
@@ -1,89 +1,89 @@
-name,server,version,luaversion,maintainers
```
Would you consider switching to `json` format or something similar ? This file is edited by humans, some other formats may be easier to edit.

teto replied:
For info, I edit the csv with either https://github.com/saulpw/visidata, libreoffice or neovim. I am not fond of json because of the absence of comments (also I like csv brevity) but I am open to changing it to json (in a follow up PR if someone is interested to do it).  I would like the `vim-plugin-names` to support more parameters like the luarocks-packages.csv . Using a similar format for both could be interesting.

teto replied:
Now that this is merged, do you want to have a go at this ? https://github.com/vaidik/commentjson/ looks popular enough.
I've also opened an issue https://github.com/NixOS/nixpkgs/issues/136248

SuperSandro2000 commented on pkgs/development/interpreters/lua-5/build-lua-package.nix:
```diff
@@ -7,9 +7,10 @@
 {
-name ? "${attrs.pname}-${attrs.version}"
+# name ? "${attrs.pname}-${attrs.version}"
+pname,
```
```suggestion
pname
```

SuperSandro2000 commented on pkgs/development/interpreters/lua-5/build-lua-package.nix:
```diff
@@ -7,9 +7,10 @@
+pname,
 
-, version
+version
```
```suggestion
, version
```

Commit:
luaPackages.ltermbox: removed

official repo here https://github.com/LuaDist-testing/ltermbox but nothing is using it and it is outdated/unmaintained (2018)
and poses issue updating

Commit:
luaPackages.luadoc: remove since last update in 2013

no update in 7 years, link towards source doesn't work anymore https://luarocks.org/modules/luarocks/luadoc/3.0.1-1

Commit:
lua: restore knownRockspec and lgi

- fix std._debug build
- fixed std-normalize
- pinned stdlib since the git version  causes issues with awesome

cleanup

Commit:
luaPackages.ansicolors: remove because unmaintained since 2013

luarocks points at https://github.com/kikito/ansicolors.lua with a last
commit in 2013 while its upstream https://github.com/hoelzro/ansicolors
contains a 2017 commit. This last project mentions it's superseeded by
https://github.com/hoelzro/lua-term

Commit:
luarocks-nix: bumped luarocks-nix

this new version removes support of src.rock but now supports rockspecs embedded in a repo. The goal is to be able to support neovim plugins that provide a rockspec, even when they are not registered on luarocks.

teto commented:
I've removed ansicolors/luadoc/ltermbox since those were posing problems and seem unused / unmaintained anyway.
I've removed src.rock support to simplify the code (how ironic since that was the first thing supported). It's now possible to reference the rockspec in a fetch archive directly.

Commit:
buildLuarocksPackage: cleanup

teto commented:
the nixpkgs-review looks ok (plex-media-player doesn't seem in cache )
```
10 packages marked as broken and skipped:
lua51Packages.digestif lua52Packages.digestif lua52Packages.gitsigns-nvim lua53Packages.gitsigns-nvim lua53Packages.ljsyscall lua53Packages.luabitop luaPackages.digestif luaPackages.gitsigns-nvim luajitPackages.digestif luajitPackages.luaffi

1 package failed to build:
plex-media-player

384 packages built:
alpine-make-vm-image anki ankisyncd apk-tools awesome celluloid curseradio d-seams gnvim haruna hydrus jellyfin-media-player jellyfin-mpv-shim jftui knot-resolver lua51Packages.alt-getopt lua51Packages.argparse lua51Packages.basexx lua51Packages.binaryheap lua51Packages.bit32 lua51Packages.busted lua51Packages.cassowary lua51Packages.cjson lua51Packages.compat53 lua51Packages.cosmo lua51Packages.coxpcall lua51Packages.cqueues lua51Packages.cyrussasl lua51Packages.dkjson lua51Packages.fifo lua51Packages.gitsigns-nvim lua51Packages.http lua51Packages.inspect lua51Packages.ldbus lua51Packages.ldoc lua51Packages.lgi lua51Packages.linenoise lua51Packages.ljsyscall lua51Packages.lpeg lua51Packages.lpeg_patterns lua51Packages.lpeglabel lua51Packages.lpty lua51Packages.lrexlib-gnu lua51Packages.lrexlib-pcre lua51Packages.lrexlib-posix lua51Packages.lua-cmsgpack lua51Packages.lua-iconv lua51Packages.lua-lsp lua51Packages.lua-messagepack lua51Packages.lua-resty-http lua51Packages.lua-resty-jwt lua51Packages.lua-resty-openidc lua51Packages.lua-resty-openssl lua51Packages.lua-resty-session lua51Packages.lua-term lua51Packages.lua-toml lua51Packages.lua-yajl lua51Packages.lua-zlib lua51Packages.lua_cliargs lua51Packages.luabitop lua51Packages.luacheck lua51Packages.luacov lua51Packages.luadbi lua51Packages.luadbi-mysql lua51Packages.luadbi-postgresql lua51Packages.luadbi-sqlite3 lua51Packages.luaepnf lua51Packages.luaevent lua51Packages.luaexpat lua51Packages.luaffi lua51Packages.luafilesystem lua51Packages.lualogging lua51Packages.luaossl lua51Packages.luaposix lua51Packages.luarepl lua51Packages.luarocks-nix lua51Packages.luasec lua51Packages.luasocket lua51Packages.luasql-sqlite3 lua51Packages.luassert lua51Packages.luasystem lua51Packages.luautf8 lua51Packages.luazip lua51Packages.luuid lua51Packages.luv lua51Packages.luxio lua51Packages.lyaml lua51Packages.markdown lua51Packages.mediator_lua lua51Packages.moonscript lua51Packages.mpack lua51Packages.nvim-client lua51Packages.penlight lua51Packages.plenary-nvim lua51Packages.rapidjson lua51Packages.readline lua51Packages.say lua51Packages.std-_debug lua51Packages.std-normalize lua51Packages.stdlib lua51Packages.vstruct luaPackages.alt-getopt luaPackages.argparse luaPackages.basexx luaPackages.binaryheap luaPackages.bit32 luaPackages.busted luaPackages.cassowary luaPackages.cjson luaPackages.compat53 luaPackages.cosmo luaPackages.coxpcall luaPackages.cqueues luaPackages.cyrussasl luaPackages.dkjson luaPackages.fifo luaPackages.http luaPackages.inspect luaPackages.ldbus luaPackages.ldoc luaPackages.lgi luaPackages.linenoise luaPackages.ljsyscall luaPackages.lpeg luaPackages.lpeg_patterns luaPackages.lpeglabel luaPackages.lpty luaPackages.lrexlib-gnu luaPackages.lrexlib-pcre luaPackages.lrexlib-posix luaPackages.lua-cmsgpack luaPackages.lua-iconv luaPackages.lua-lsp luaPackages.lua-messagepack luaPackages.lua-resty-http luaPackages.lua-resty-jwt luaPackages.lua-resty-openidc luaPackages.lua-resty-openssl luaPackages.lua-resty-session luaPackages.lua-term luaPackages.lua-toml luaPackages.lua-yajl luaPackages.lua-zlib luaPackages.lua_cliargs luaPackages.luabitop luaPackages.luacheck luaPackages.luacov luaPackages.luadbi luaPackages.luadbi-mysql luaPackages.luadbi-postgresql luaPackages.luadbi-sqlite3 luaPackages.luaepnf luaPackages.luaevent luaPackages.luaexpat luaPackages.luaffi luaPackages.luafilesystem luaPackages.lualogging luaPackages.luaossl luaPackages.luaposix luaPackages.luarepl luarocks-nix luaPackages.luasec luaPackages.luasocket luaPackages.luasql-sqlite3 luaPackages.luassert luaPackages.luasystem luaPackages.luautf8 luaPackages.luazip luaPackages.luuid luaPackages.luv luaPackages.luxio luaPackages.lyaml luaPackages.markdown luaPackages.mediator_lua luaPackages.moonscript luaPackages.mpack luaPackages.nvim-client luaPackages.penlight luaPackages.plenary-nvim luaPackages.rapidjson luaPackages.readline luaPackages.say luaPackages.std-_debug luaPackages.std-normalize luaPackages.stdlib luaPackages.vstruct lua53Packages.alt-getopt lua53Packages.argparse lua53Packages.basexx lua53Packages.binaryheap lua53Packages.bit32 lua53Packages.busted lua53Packages.cassowary lua53Packages.cjson lua53Packages.compat53 lua53Packages.cosmo lua53Packages.coxpcall lua53Packages.cqueues lua53Packages.cyrussasl lua53Packages.digestif lua53Packages.dkjson lua53Packages.fifo lua53Packages.http lua53Packages.inspect lua53Packages.ldbus lua53Packages.ldoc lua53Packages.lgi lua53Packages.linenoise lua53Packages.lpeg lua53Packages.lpeg_patterns lua53Packages.lpeglabel lua53Packages.lpty lua53Packages.lrexlib-gnu lua53Packages.lrexlib-pcre lua53Packages.lrexlib-posix lua53Packages.lua-cmsgpack lua53Packages.lua-iconv lua53Packages.lua-lsp lua53Packages.lua-messagepack lua53Packages.lua-resty-http lua53Packages.lua-resty-jwt lua53Packages.lua-resty-openidc lua53Packages.lua-resty-openssl lua53Packages.lua-resty-session lua53Packages.lua-term lua53Packages.lua-toml lua53Packages.lua-yajl lua53Packages.lua-zlib lua53Packages.lua_cliargs lua53Packages.luacheck lua53Packages.luacov lua53Packages.luadbi lua53Packages.luadbi-mysql lua53Packages.luadbi-postgresql lua53Packages.luadbi-sqlite3 lua53Packages.luaepnf lua53Packages.luaevent lua53Packages.luaexpat lua53Packages.luaffi lua53Packages.luafilesystem lua53Packages.lualogging lua53Packages.luaossl lua53Packages.luaposix lua53Packages.luarepl lua53Packages.luarocks-nix lua53Packages.luasec lua53Packages.luasocket lua53Packages.luasql-sqlite3 lua53Packages.luassert lua53Packages.luasystem lua53Packages.luautf8 lua53Packages.luazip lua53Packages.luuid lua53Packages.luv lua53Packages.luxio lua53Packages.lyaml lua53Packages.markdown lua53Packages.mediator_lua lua53Packages.moonscript lua53Packages.mpack lua53Packages.nvim-client lua53Packages.penlight lua53Packages.plenary-nvim lua53Packages.rapidjson lua53Packages.readline lua53Packages.say lua53Packages.std-_debug lua53Packages.std-normalize lua53Packages.stdlib lua53Packages.vstruct luajitPackages.alt-getopt luajitPackages.argparse luajitPackages.basexx luajitPackages.binaryheap luajitPackages.bit32 luajitPackages.busted luajitPackages.cassowary luajitPackages.cjson luajitPackages.compat53 luajitPackages.cosmo luajitPackages.coxpcall luajitPackages.cqueues luajitPackages.cyrussasl luajitPackages.dkjson luajitPackages.fifo luajitPackages.gitsigns-nvim luajitPackages.http luajitPackages.inspect luajitPackages.ldbus luajitPackages.ldoc luajitPackages.lgi luajitPackages.linenoise luajitPackages.ljsyscall luajitPackages.lpeg luajitPackages.lpeg_patterns luajitPackages.lpeglabel luajitPackages.lpty luajitPackages.lrexlib-gnu luajitPackages.lrexlib-pcre luajitPackages.lrexlib-posix luajitPackages.lua-cmsgpack luajitPackages.lua-iconv luajitPackages.lua-lsp luajitPackages.lua-messagepack luajitPackages.lua-resty-http luajitPackages.lua-resty-jwt luajitPackages.lua-resty-openidc luajitPackages.lua-resty-openssl luajitPackages.lua-resty-session luajitPackages.lua-term luajitPackages.lua-toml luajitPackages.lua-yajl luajitPackages.lua-zlib luajitPackages.lua_cliargs luajitPackages.luabitop luajitPackages.luacheck luajitPackages.luacov luajitPackages.luadbi luajitPackages.luadbi-mysql luajitPackages.luadbi-postgresql luajitPackages.luadbi-sqlite3 luajitPackages.luaepnf luajitPackages.luaevent luajitPackages.luaexpat luajitPackages.luafilesystem luajitPackages.lualogging luajitPackages.luaossl luajitPackages.luaposix luajitPackages.luarepl luajitPackages.luarocks-nix luajitPackages.luasec luajitPackages.luasocket luajitPackages.luasql-sqlite3 luajitPackages.luassert luajitPackages.luasystem luajitPackages.luautf8 luajitPackages.luazip luajitPackages.luuid luajitPackages.luv luajitPackages.luxio luajitPackages.lyaml luajitPackages.markdown luajitPackages.mediator_lua luajitPackages.moonscript luajitPackages.mpack luajitPackages.nvim-client luajitPackages.penlight luajitPackages.plenary-nvim luajitPackages.rapidjson luajitPackages.readline luajitPackages.say luajitPackages.std-_debug luajitPackages.std-normalize luajitPackages.stdlib luajitPackages.vstruct luakit minitube mnemosyne mpc-qt mpv mpv-unwrapped mpvScripts.mpris mudlet neovim-qt neovim-qt-unwrapped neovim-remote neovim-unwrapped nvimpager openrussian-cli plex-mpv-shim prosody python38Packages.mpv python39Packages.mpv qimgv quvi sile somafm-cli sublime-music telegram-cli vis weechatScripts.weechat-matrix-bridge wordgrinder ytfzf z-lua
```
this has the potential to break stuff so I am going to check some more software and may merge afterwards (e.g. >= tomorrow).

teto commented:
I've tested neovim/sile/ytfzf/anki that seem to work fine. Merging this and let's hope it doesn't break too many setups. If you are using one of the removed packages, we can discuss a solution together.
